### ISSUE TITLE
Item () of () should be able to be chosen "random","last"

### ISSUE THREAD

**Issue by apple502j** [LLK/scratch-blocks#1335]
### Expected Behavior
item () of ()
insert () at () of ()
replace item () of () with ()

They should be able to be chosen "random","last" 
### Actual Behavior
It allows only numbers
### Steps to Reproduce
1. Make a list.
2. See the block.
### Operating System and Browser
Windows 7 SP1 64bit HomePremium, Firefox 57.0.4

**Comment by ZenithRogue** [comment]
It would be nice to see the dropdown added to the (letter () of []) blocks as well

**Comment by Kenny2github** [comment]
Perhaps allow negative indices? That would allow for indices a certain distance from the end :)

**Comment by lawsie** [comment]
To add to this, it is actually now very difficult to add anything to the end of a list.
(Edit: I was actually wrong but it's still quite unintuitive for beginners).

![example2](https://user-images.githubusercontent.com/5183697/49803952-3f39e980-fd49-11e8-8ca4-de08f65980da.png)


This has caused us an issue with one of the Raspberry Pi Scratch projects.

**Comment by towerofnix** [comment]
@lawsie I'm a little confused, why not just use the block "add (Test) to (Example list)"?

**Comment by lawsie** [comment]
Haha @towerofnix that would certainly work - that's me over complicating it! Still, you'd need to use that syntax to find the item at the end of the list now, unless I'm mistaken?

![end of list](https://user-images.githubusercontent.com/5183697/49824630-7248a100-fd7a-11e8-9990-8ce26ea94726.png)

**Comment by towerofnix** [comment]
Yeah, @lawsie, that's right. It's the same for "random" as well, i.e. item (pick random (1) to (length of (list)) of (list). (And you have to use that same "pick random" for any of the other list blocks too. Likewise, you have to do "delete (length of (example list)) of (example list)" to remove the last item from the list.)

**Comment by lawsie** [comment]
Right. Well even though my previous example was invalid, that's still not as easy as (item (last v) of [list v]) for a child to understand, so I'm hoping that last and random will be re-added. :)

**Comment by joker314** [comment]
Just to note that this is _not_ an urgent compatibility issue because the VM understands what "last" and "random" mean, it's just the user interface won't let you type it in for new projects. (You can see this by using a 'join' block to enter random/last)

Still would be awesome, especially since 2.0 had it, and I agree with @lawsie 's point that it isn't intuitive.

**Comment by lawsie** [comment]
Thanks, and yes, where our (Raspberry Pi) projects use lists we have been able to successfully change the syntax to something that works for release day, but if last/random get re-added we will definitely change it back for clarity/pedagogy reasons.

**Comment by cwillisf** [comment]
This is a feature request for the Scratch language or runtime rather than for the scratch-blocks editor. scratch-blocks handles the visual block editor; changes to the block library, language, or runtime would be tracked elsewhere. Closing as part of an issue cleanup. Thank you for the suggestion!

If you believe this issue is still relevant to scratch-blocks v2, please let us know and we can reopen it. It may also be appropriate to open a new issue in the [scratch-editor](https://github.com/scratchfoundation/scratch-editor) repository instead.
